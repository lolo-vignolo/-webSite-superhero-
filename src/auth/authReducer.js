import React from 'react'
import { types } from '../types/types';

//estado si el usuario esta autenticado
// const state = {
//     name: "Lorenzo",
//     logged: true
// }


// const ejAction = {

//     type : "login",
//     payload : {
//         name : "newname",
//         logged : true or false
//     }
// }


export const authReducer = (state = {}, action ) => {
   

    switch (action.type) {
        case  types.login:
            
            return {
              ...action.payload,
              logged : true
          }
          
        case types.logout:
            
            return{
                logged:false
            }

        default:
        return state;



    }
}

