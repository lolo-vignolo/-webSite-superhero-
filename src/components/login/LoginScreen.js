import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';


export const LoginScreen = () => {

    const navigate = useNavigate();
    const { dispatch } = useContext( AuthContext )

    const handleLogin = () => {
        const action = {
            type: types.login,
            payload: { name: 'Lorenzo' }
        }

        dispatch(action);

        const lastPath = localStorage.getItem('lastPath') || '/marvel';


        navigate( lastPath, {
            replace: true
        });
    }


    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr />

            <button 
                className="btn btn-secondary"
                onClick={ handleLogin }
                >
                    Login
            </button>
        </div>
    )
}
// import React, { useContext } from 'react';
// import { AuthContext } from '../../auth/AuthContext';


// import { types } from '../../types/types';




// export const LoginScreen = ({history}) => {

//     const lastPath = localStorage.getItem("lastPath") || "/";


//     const {name} = {
//         name:"Lorenzo"
//     }
   
//     const {dispatch} = useContext(AuthContext);

  
//     const handleLogin = ()=>{
//         // history.push("/"); 
       
//         const login = {
//             type : types.login,
//             payload:{name}
//         }

//         dispatch(login)
//         history.replace(lastPath);
//     } 


//     return (
//         <div className = "container mt-5">
//             <h1>Login</h1>
//             <hr />
        
//             <button 
//             className= "btn btn-secondary"
//             onClick = {handleLogin}
//             >
//                Login  
//             </button>
//         </div>
//     )
// }
