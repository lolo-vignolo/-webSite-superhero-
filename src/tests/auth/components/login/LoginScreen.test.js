import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../../../auth/AuthContext";
import { LoginScreen } from "../../../../components/login/LoginScreen";
import { types } from "../../../../types/types";

describe('prueba en <LoginScreen />', () => {
    test('debe mostrarse correctamente', () => {
        const history = {
            
            replace: jest.fn(),
        }

        const contextValue = {
            dispatch: jest.fn(),
            user:{
                logged: true,
                name : "Lorenzo"
            }
        }

        const wrapper = mount (
            <AuthContext.Provider value = {contextValue} >
            <MemoryRouter >

                <LoginScreen history = {history}/>
              
            </MemoryRouter>
            </AuthContext.Provider>
        
        )

        expect(wrapper).toMatchSnapshot();

        wrapper.find("button").prop("onClick")();
        expect(contextValue.dispatch).toHaveBeenCalled();
        expect(history.replace).toHaveBeenCalled();

        //para testear con local storage. ==>

        // localStorage.setItem("lastPath", "/dc")
        // wrapper.find("button").prop("onClick")();
        // expect(history.replace).toHaveBeenCalledWith("/dc");
                
    });


    
    
})
