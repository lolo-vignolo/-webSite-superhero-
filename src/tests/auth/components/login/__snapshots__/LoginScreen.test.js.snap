// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en <LoginScreen /> debe mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
